packages/blade/src/components/Radio/RadioGroup/useRadioGroup.ts

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import isUndefined from 'lodash/isUndefined';
+import type { RadioGroupProps } from './RadioGroup';
+import type { RadioGroupContextType } from './RadioContext';
+import { useControllableState } from '~src/hooks/useControllable';
+import { useTheme } from '~components/BladeProvider';
+import { useId } from '~src/hooks/useId';
+
+type UseRadioGroupProps = Pick<
+  RadioGroupProps,
+  | 'isDisabled'
+  | 'labelPosition'
+  | 'validationState'
+  | 'name'
+  | 'value'
+  | 'defaultValue'
+  | 'onChange'
+  | 'neccessityIndicator'
+>;
+
+export type State = {
+  value: string;
+  setValue(value: string): void;
+  removeValue(): void;
+  isChecked(value: string): boolean;
+};
+
+type UseRadioGroupReturn = {
+  state: State;
+  contextValue: RadioGroupContextType;
+  ids: { labelId: string };
+};
+
+const useRadioGroup = ({
+  value,
+  defaultValue,
+  isDisabled,
+  labelPosition,
+  onChange,
+  validationState,
+  neccessityIndicator,
+  name,
+}: UseRadioGroupProps): UseRadioGroupReturn => {
+  const { platform } = useTheme();
+  const idBase = useId('radio-group');
+  const labelId = `${idBase}-label`;
+  const fallbackName = name ?? idBase;
+  const [checkedValue, setValue] = useControllableState({
+    value,
+    defaultValue,
+    onChange: (v: string) => onChange?.({ value: v, name: fallbackName }),
+  });
+
+  // FIX React native bug, where it unchecks the radio on clicking it again
+  const state = React.useMemo<State>(() => {
+    return {
+      value: checkedValue,
+      setValue(v: string): void {
+        if (isDisabled) {
+          return;
+        }
+
+        setValue(() => v);
+      },
+      removeValue(): void {
+        if (isDisabled) {
+          return;
+        }
+
+        // @ts-expect-error TODO: fix this
```

Comment:
Looks like a pending todo. Can you take a look?

Reply:
Just a TS error. setValue() doesn't accept undefined but I want it to, to be able to remove the selected radio. Should be fine.